Source/Initialization/WarpXInitData.cpp

```diff
@@ -747,6 +747,25 @@ WarpX::InitLevelData (int lev, Real /*time*/)
                    E_ext_grid_s.begin(),
                    ::tolower);
 
+    // Externally imposed fields are only initialized until the user-defined maxlevel_extEMfield_init.
+    // The default maxlevel_extEMfield_init value is the total number of levels in the simulation
+    // Note that if the external fields are not provided, the fields, E and B, on all levels
+    // will still be initialized with the default value of 0.
+    if ( (B_ext_grid_s == "constant") || (B_ext_grid_s == "parse_b_ext_grid_function")) {
```

Comment:
I would suggest that this should be
```if (B_ext_grid_s != "default" && lev > maxlevel_extEMfield_init) {```
For the same reason above, in case a new method is added, and to simplify the nested `if` statements.

A general comment, this bit of code could be removed and the logic below simplified if the E and B fields were initialized when allocated, by passing in zero as the initial value when `AllocInitMultiFab` is called for these MultiFabs. This would be more robust anyway to mistakes that could leave the E and B uninitialized.

Reply:
This is now fixed.